nexus/db-macros/src/lookup.rs

```diff
@@ -287,63 +284,70 @@ fn generate_struct(config: &Config) -> TokenStream {
     }
 }
 
-/// Generates the child selectors for this resource
+/// Generates the child selector for this resource's parent
 ///
-/// For example, for the "Project" resource with child resources "Instance" and
-/// "Disk", this will generate the `Project::instance_name()` and
-/// `Project::disk_name()` functions.
-fn generate_child_selectors(config: &Config) -> TokenStream {
-    let child_resource_types: Vec<_> =
-        config.child_resources.iter().map(|c| format_ident!("{}", c)).collect();
-    let child_selector_fn_names: Vec<_> = config
-        .child_resources
-        .iter()
-        .map(|c| format_ident!("{}_name", heck::AsSnakeCase(c).to_string()))
-        .collect();
-    let child_selector_fn_names_owned: Vec<_> = config
-        .child_resources
-        .iter()
-        .map(|c| {
-            format_ident!("{}_name_owned", heck::AsSnakeCase(c).to_string())
-        })
-        .collect();
-    let child_selector_fn_docs: Vec<_> = config
-        .child_resources
-        .iter()
-        .map(|child| {
-            format!(
-                "Select a resource of type {} within this {}, \
-                identified by its name",
-                child, config.resource.name,
-            )
-        })
-        .collect();
+/// For example, for the "Instance" resource with parent resource "Project",
+/// this will generate the `Project::instance_name()` and
+/// `Project::instance_name_owned()` functions.
+///
+/// This is generated by the child resource codegen, rather than by the parent
+/// resource codegen, since such functions are only generated for resources with
+/// `lookup_by_name = true`.  Whether or not this is enabled for the child
+/// resource is not known when generating the parent resource code, so it's
+/// performed by the child instead.
+fn generate_child_selector(config: &Config) -> TokenStream {
+    // If this resource can only be looked up by ID, we don't need to generate
+    // child selectors on the parent resource.
+    if !config.lookup_by_name {
+        return quote! {};
+    }
```

Comment:
To confirm: This is like, the "functional crux" of the change, the rest is moving the generation from "generated in parent declaration" to "generated in child declaration", which is more of a refactor.

Reply:
Yeah, that's correct. Skipping this generation when the child resource can only be looked up by ID is necessary to stop generating code that won't compile for such resources. Moving around where that code gets generated is only necessary because otherwise, we wouldn't have the _information_ about whether the child is looked up by name.